app/controllers/liststreams_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ListstreamsController < ApplicationController
+
+  @limit = 5
+  
+  def list_streams
+    begin
+      @client = ListstreamsHelper.create_aws_client session[:user_id]
+      @resp = ListstreamsHelper.build_response(@client, @limit)
+      redirect_to({:action => "index"}, notice: @resp)
+    rescue UnableToRescueAwsResponse => e
+      @resp.errors.add(:base, message: e.message)
+      render :index, status: :unprocessable_entity
+    rescue AwsCredentialValidationError => e
+      @client.errors.add(:base, message: e.message)
+      render :index, status: :unprocessable_entity
+    end
+  end
+
+  def index
+    flash.clear
+  end
```

Comment:
Please don't do that, as then the page won't be able to render the streams because they were stored on flash.notice and all of flash' content is being cleaned before the view is generated.
---
What we can do is the following:
```ruby
def index
  flash.keep(:streams)
end
```
And instead of line 9, we do the following: 
```ruby
flash[:streams] = @resp
redirect_to({:action => "index"})
```
And in the view we change `flash.notice` by `flash[:streams]`
---
This way we make sure that the view is always getting an specific entry of the flash hash, the `:streams` entry